File: client/types/contestant.ts
export interface Contestant {
  id: string;
  name: string;
  dateOfBirth: string;
  photoURL?: string;
  gender: string;
  academicYear: string;
  semester: string;
  nic?: string;
  studentId?: string;
  createdAt: string;
  updatedAt: string;
}

export interface ContestantInput {
  name: string;
  dateOfBirth: string;
  photoURL?: string;
  gender: string;
  academicYear: string;
  semester: string;
  nic?: string;
  studentId?: string;
}

export interface ContestantListPagination {
  page: number;
  limit: number;
  total: number;
  totalPages: number;
  hasNext: boolean;
  hasPrev: boolean;
}

export interface PublicContestantListResponse {
  contestants: Contestant[];
  pagination: ContestantListPagination;
}

File: server/internal/model/domain/contestant.go
package domain

import "time"

type Contestant struct {
	ID             string    `json:"id"`
	Name           string    `json:"name"`
	Birthday       *string   `json:"birthday"`
	NicOrStudentID string    `json:"nicOrStudentId"`
	PhotoURL       *string   `json:"photoUrl"`
	Gender         *string   `json:"gender"`
	AcademicYear   *string   `json:"academicYear"`
	CreatedAt      time.Time `json:"createdAt"`
	UpdatedAt      time.Time `json:"updatedAt"`
}
